packages/tailwindcss/src/variants.ts

```diff
@@ -901,8 +901,10 @@ export function createVariants(theme: Theme): Variants {
 function quoteAttributeValue(value: string) {
   if (value.includes('=')) {
     value = value.replace(/(=.*)/g, (_fullMatch, match) => {
+      let value = match.slice(1).trim()
+
       // If the value is already quoted, skip.
-      if (match[1] === "'" || match[1] === '"') {
+      if (value[0] === "'" || value[0] === '"') {
```

Comment:
`_` values in the value are already converted to ` ` which means that we can just trim the valid (if we ignore the first `=` character)